prov/rxm/src/rxm_cq.c

```diff
@@ -1356,9 +1356,7 @@ void rxm_ep_do_progress(struct util_ep *util_ep)
 	if (!slistfd_empty(&rxm_ep->msg_eq_entry_list))
 		rxm_conn_process_eq_events(rxm_ep);
 
-	while (!dlist_empty(&rxm_ep->repost_ready_list)) {
-		dlist_pop_front(&rxm_ep->repost_ready_list, struct rxm_rx_buf,
-				buf, repost_entry);
+	while (!rxm_rx_buf_acquire(rxm_ep, &buf)) {
```

Comment:
I don't understand the motivation for this change.  It's a slight restructuring of the code, but less efficient than the previous code.  (Assuming that the compiler doesn't just inline everything.

Reply:
@shefty Hmm, taking into account the possible issue with inlining you pointed above and fact that we try to acquire buffer from repost list only in this function, maybe we don't need rxm_rx_buf_acquire() at all.